src/components/PasswordField/PasswordField.vue

```diff
@@ -0,0 +1,318 @@
+<!--
+  - @copyright Copyright (c) 2022 Marco Ambrosini <marcoambrosini@pm.me>
+  -
+  - @author Marco Ambrosini <marcoambrosini@pm.me>
+  - @author Carl Schwan <carl@carlschwan.eu>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<docs>
+### Description
+
+General purpose password field component.
+
+```
+<template>
+	<div class="wrapper">
+		<PasswordField :value.sync="text1"
+			label="Old password" />
+		<div class="external-label">
+			<label for="$refs.textField.id">New password</label>
+			<PasswordField :value.sync="text2"
+				ref="textField"
+				:label-outside="true" />
+		</div>
+		<div class="external-label">
+			<label for="textField2">New password</label>
+			<PasswordField :value.sync="text3"
+				:error="true"
+				id="textField2"
+				:label-outside="true"
+				helper-text="Password is insecure" />
+		</div>
+
+		<PasswordField :value.sync="text4"
+			label="Good new password"
+			label-visible="true"
+			:success="true"
+			id="textField2"
+			helper-text="Password is secure" />
+	</div>
+</template>
+<script>
+export default {
+	data() {
+		return {
+			text1: '',
+			text2: '',
+			text3: '',
+			text4: '',
+		}
+	},
+}
+</script>
+<style lang="scss" scoped>
+.wrapper {
+	display: flex;
+	gap: 4px;
+	align-items: flex-end;
+	flex-wrap: wrap;
+}
+
+.external-label {
+	display: flex;
+	width: 100%;
+	margin-top: 1rem;
+}
+</style>
+```
+</docs>
+
+<template>
+	<InputField v-bind="$props"
+		:type="isPasswordHidden ? 'password' : 'text'"
+		:show-trailing-button="true"
+		:helper-text="computedHelperText"
+		v-on="$listeners"
+		@trailing-button-click="togglePasswordVisibility"
+		@input="handleInput">
+		<!-- Default slot for the leading icon -->
+		<slot />
+		<template slot="trailing-button-icon">
+			<Eye v-if="isPasswordHidden" :size="18" />
+			<EyeOff v-else :size="18" />
+		</template>
+	</InputField>
+</template>
+
+<script>
+
+import Eye from 'vue-material-design-icons/Eye'
+import EyeOff from 'vue-material-design-icons/EyeOff'
+import InputField from '../InputField/InputField.vue'
+import _ from 'lodash'
```

Comment:
We should avoid to import full `lodash` package.
Instead of it, we can import only what we need.

```js
import { debounce } from 'lodash'
```

Besides that, it is highly recommended use [`lodash-es`](https://www.npmjs.com/package/lodash-es) instead of `lodash` .
It will make possible [`tree shaking`](https://webpack.js.org/guides/tree-shaking/) and improve the bundle size.